jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -1450,8 +1457,11 @@ ecma_regexp_run (ecma_regexp_ctx_t *re_ctx_p, /**< RegExp matcher context */
 
         const lit_code_point_t cp = ecma_regexp_unicode_advance (&str_curr_p, re_ctx_p->input_end_p);
 
-        if (JERRY_UNLIKELY (cp <= LIT_UTF16_CODE_UNIT_MAX && lit_char_is_line_terminator ((ecma_char_t) cp)))
+        if (JERRY_UNLIKELY (cp <= LIT_UTF16_CODE_UNIT_MAX &&
```

Comment:
if `(re_ctx_p->flags & RE_FLAG_DOTALL)` present, the remaining conditions should not be evaluated.